fetchmedia.go

```diff
@@ -70,8 +96,14 @@ func fetchMedia(mediaSource, mediaFolderPath, mediaFilename string) (mediaType s
 	return mtype, nil
 }
 
-func fetchMediaFromURL(mediaSource string) (io.ReadCloser, error) {
-	resp, err := http.Get(mediaSource)
+func (g grabber) httpHandler(mediaSource string, onlyCheck bool) (io.ReadCloser, error) {
+	var resp *http.Response
+	var err error
+	if onlyCheck {
+		resp, err = g.Head(mediaSource)
```

Comment:
I love this! very nice work using a head request when we don't need to do a full get.